app/pages/project/instances/instance/tabs/SettingsTab.tsx

```diff
@@ -0,0 +1,147 @@
+ *
+ * Copyright Oxide Computer Company
+ */
+
+import { format, formatDistanceToNow } from 'date-fns'
+import { useEffect, useId, useState, type ReactNode } from 'react'
+import { useForm } from 'react-hook-form'
+
+import { apiQueryClient, useApiMutation, usePrefetchedApiQuery } from '~/api'
+import { ListboxField } from '~/components/form/fields/ListboxField'
+import { useInstanceSelector } from '~/hooks/use-params'
+import { addToast } from '~/stores/toast'
+import { Button } from '~/ui/lib/Button'
+import { FieldLabel } from '~/ui/lib/FieldLabel'
+import { LearnMore, SettingsGroup } from '~/ui/lib/SettingsGroup'
+import { TipIcon } from '~/ui/lib/TipIcon'
+import { links } from '~/util/links'
+
+Component.displayName = 'SettingsTab'
+export function Component() {
+  const instanceSelector = useInstanceSelector()
+
+  const [now, setNow] = useState(new Date())
+
+  useEffect(() => {
+    const timer = setInterval(() => setNow(new Date()), 1000)
+    return () => clearInterval(timer)
+  }, [])
+
+  const { data: instance } = usePrefetchedApiQuery('instanceView', {
+    path: { instance: instanceSelector.instance },
+    query: { project: instanceSelector.project },
+  })
+
+  const instanceUpdate = useApiMutation('instanceUpdate', {
+    onSuccess() {
+      apiQueryClient.invalidateQueries('instanceView')
+      addToast({ content: 'Instance auto-restart policy updated' })
+    },
+  })
+
+  const form = useForm({
+    defaultValues: {
+      autoRestartPolicy: instance.autoRestartPolicy,
+    },
+  })
+  const { isDirty } = form.formState
+
+  const onSubmit = form.handleSubmit((values) => {
+    instanceUpdate.mutate({
+      path: { instance: instanceSelector.instance },
+      query: { project: instanceSelector.project },
+      body: {
+        ncpus: instance.ncpus,
+        memory: instance.memory,
+        bootDisk: instance.bootDiskId,
+        autoRestartPolicy: values.autoRestartPolicy,
+      },
+    })
+  })
+
+  return (
+    <form className="space-y-6" onSubmit={onSubmit}>
+      <SettingsGroup.Container>
+        <SettingsGroup.Header>
+          <SettingsGroup.Title>Auto-restart</SettingsGroup.Title>
+          <p>The auto-restart policy configured for this instance</p>
+        </SettingsGroup.Header>
+        <SettingsGroup.Body>
+          <ListboxField
+            control={form.control}
+            name="autoRestartPolicy"
+            label="Policy"
+            description="If unconfigured this instance uses the default auto-restart policy, which may or may not allow it to be restarted."
+            placeholder="Default"
+            items={[
+              { value: '', label: 'None (Default)' },
+              { value: 'never', label: 'Never' },
+              { value: 'best_effort', label: 'Best effort' },
+            ]}
+            required
+            className="max-w-none"
+          />
+          <FormMeta label="Enabled" helpText="Help">
+            {instance.autoRestartEnabled ? 'True' : 'False'}{' '}
```

Comment:
With our mock instance, it's possible for the policy to be set to never but `autoRestartEnabled` to be true. In case of the actual API it'll automatically set `autoRestartEnabled` to be false.

https://github.com/oxidecomputer/omicron/blob/f63ed095e744fb8d2383fda6799eb0b2d6dfbd3c/nexus/db-queries/src/db/datastore/instance.rs#L228C26-L239

We might want to modify the mock `InstanceView` handler to match this behaviour.

Reply:
Yeah, definitely modify the mock handler and link to the API code in a comment.